docs/sphinx/tutorial.rst

```diff
@@ -0,0 +1,66 @@
+
+It should also be noted that there is a gpu_simulation mode. It is generally much harder to debug issues on the device, so we have a mode that runs the code on the host, and if CHAI is configured with -DCHAI_ENABLE_GPU_SIMULATION_MODE=ON, the "GPU" memory space is actually on the host as well. This enables some stale data to be caught more easily when CHAI is used as a memory manager, and also enables address sanitizer or valgrind to be used on the simulated GPU memory space. CARE must also be configured with -DCARE_ENABLE_GPU_SIMULATION_MODE=ON.
+
+Another policy of note is raja_chai_everywhere. This runs the code on both the host and the device and is intended to be used with managed_ptr to keep both the host and device object instance in sync.
+
+Finally, there are fusible policies. Loops can be "fused" together into a single kernel to reduce kernel launch overhead. More details will be forthcoming on this.
+
+These policies can be used with care::forall, which then forwards the arguments to RAJA::forall. CARE also provides convenient macros that can fall back to vanilla for loops if CARE is configured with -DENABLE_LEGACY_CARE=ON. These macros generally start with CARE and end with LOOP. For example, CARE_SEQUENTIAL_LOOP always executes sequentially. CARE_STREAM_LOOP executes on the GPU if GPU_ACTIVE is defined in the compilation unit and nvcc is used to compile the code, otherwise it falls back to executing using OpenMP on the host if OPENMP_ACTIVE is defined in the compilation unit and the correct OpenMP flags are passed to the compiler, otherwise it falls back to running sequentially on the host.
+
+.. code-block:: c++
+
+   #define GPU_ACTIVE // If this is not defined in the compilation unit, will not run on the device
+   #define OPENMP_ACTIVE // If this is not defined in the compilation unit, will not run with OpenMP
+
+   #include "care/care.h"
+
+   int main(int argc, char* argv[]) {
+      int length = 10;
+      care::host_device_ptr<int> array(length, "array");
+
+      // Will run on the GPU if this is compiled with nvcc.
+      // Otherwise will run sequentially on the host.
+      CARE_GPU_LOOP(i, 0, length) {
+         array[i] = i;
+      } CARE_GPU_LOOP_END
+
+      // Will run in parallel on the host if compiled with the OpenMP flags.
+      // Otherwise will run sequentially on the host.
+      CARE_OPENMP_LOOP(i, 0, length) {
+         array[i] = array[i] * i;
+      } CARE_OPENMP_LOOP_END
+
+      // Will run on the GPU if compiled with nvcc.
```

Comment:
probably should comment as compie with nvcc or hip

Reply:
Good point. Done.